addons/mail/static/src/messaging/entity/attachment/attachment.js

```diff
@@ -0,0 +1,305 @@
+            }
+            if (!attachments.includes(attachment)) {
+                return;
+            }
+            this.env.entities.Dialog.open('AttachmentViewer', { attachment, attachments });
+        }
+
+        /**
+         * @returns {string}
+         */
+        get defaultSource() {
+            const fileType = this.fileType;
+            if (fileType === 'image') {
+                return `/web/image/${this.id}?unique=1&amp;signature=${this.checksum}&amp;model=ir.attachment`;
+            }
+            if (fileType === 'application/pdf') {
+                return `/web/static/lib/pdfjs/web/viewer.html?file=/web/content/${this.id}?model%3Dir.attachment`;
+            }
+            if (fileType && fileType.indexOf('text') !== -1) {
+                return `/web/content/${this.id}?model%3Dir.attachment`;
+            }
+            if (fileType === 'youtu') {
+                if (fileType !== 'youtu') {
+                    return undefined;
+                }
+                const urlArr = this.url.split('/');
+                let token = urlArr[urlArr.length - 1];
+                if (token.indexOf('watch') !== -1) {
+                    token = token.split('v=')[1];
+                    const amp = token.indexOf('&');
+                    if (amp !== -1) {
+                        token = token.substring(0, amp);
+                    }
+                }
+                return `https://www.youtube.com/embed/${token}`;
+            }
+            if (fileType === 'video') {
+                return `/web/image/${this.id}?model=ir.attachment`;
+            }
+            return undefined;
+        }
+
+        /**
+         * @returns {string}
+         */
+        get displayName() {
+            return this.name || this.filename;
+        }
+
+        /**
+         * @returns {string}
+         */
+        get extension() {
+            return this.filename && this.filename.split('.').pop();
+        }
+
+        /**
+         * @returns {string}
+         */
+        get fileType() {
+            if (this.type === 'url' && !this.url) {
+                return undefined;
+            } else if (!this.mimetype) {
+                return undefined;
+            }
+            const match = this.type === 'url'
+                ? this.url.match('(youtu|.png|.jpg|.gif)')
+                : this.mimetype.match('(image|video|application/pdf|text)');
+            if (!match) {
+                return undefined;
+            }
+            if (match[1].match('(.png|.jpg|.gif)')) {
+                return 'image';
+            }
+            return match[1];
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isLinkedToComposer() {
+            return this.composers.length > 0;
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isTextFile() {
+            const fileType = this.fileType;
+            return (fileType && fileType.indexOf('text') !== -1) || false;
+        }
```

Comment:
I find this more readable

```suggestion
        get isTextFile() {
            if (!this.fileType) {
                return false;
            }
            return this.fileType.includes('text');
        }
```